experimental/dds/tree2/src/feature-libraries/utils.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IJsonCodec } from "../codec";
+import { RevisionTag, EncodedRevisionTag, ChangeAtomId, EncodedChangeAtomId } from "../core";
+
+export function encodeChangeAtomId(
+	revisionTagCodec: IJsonCodec<RevisionTag, EncodedRevisionTag>,
+	changeAtomId: ChangeAtomId,
+): EncodedChangeAtomId {
+	if (changeAtomId.revision === undefined) {
+		return { localId: changeAtomId.localId };
+	}
+
+	return {
+		localId: changeAtomId.localId,
+		revision: revisionTagCodec.encode(changeAtomId.revision),
+	};
+}
+
+export function decodeChangeAtomId(
+	revisionTagCodec: IJsonCodec<RevisionTag, EncodedRevisionTag>,
+	changeAtomId: EncodedChangeAtomId,
+): ChangeAtomId {
+	if (changeAtomId.revision === undefined) {
+		return { localId: changeAtomId.localId };
+	}
+
+	return {
+		localId: changeAtomId.localId,
+		revision: revisionTagCodec.decode(changeAtomId.revision),
+	};
+}
+
+export function decodeChangeAtomIdDefined(
```

Comment:
Is there a difference between this function and `decodeChangeAtomId`?

Reply:
Oops, meant to remove this. I was trying to avoid having to do a bunch of ternary expressions in sequence field before I ended up just doing it. I'll push a commit removing this now.